docs/madr/decisions/017-mesh-proxy-template.md

```diff
@@ -0,0 +1,244 @@
+          operation: patch
+          match: # optional: if absent, all clusters will be patched
+            name: test-cluster # optional: if absent, all clusters regardless of name will be patched
+            origin: inbound # optional: if absent, all clusters regardless of its origin will be patched
+          value: | # you can specify only part of cluster definition that will be merged into existing cluster
+            connectTimeout: 5s
+          jsonPatches: # either jsonPatches or value. Only possible to use with operation: patch. See https://jsonpatch.com/ for more.
+            - op: add
+              from: ""
+              path: "/connectTimeout"
+              value: "5s"
+      - cluster:
+          operation: remove
+          match: # optional: if absent, all clusters will be removed
+            name: test-cluster # optional: if absent, all clusters regardless of name will be removed
+            origin: inbound # optional: if absent, all clusters regardless of its origin will be removed
+```
+
+A couple of things to note when you compare this to the old policy:
+* `imports` is gone, solving _Profile imports section_ problem.
+* `appendModifications` allows to append modifications. This solves _No aggregation_ problem.
+* Because list are appended, there is no way to override or exclude modifications.
+  This should be fine for this time being, since proxy templates are managed by the mesh operator.
+  We could introduce `modifications` next to `appendModifications` or provide more general mechanism for excluding/overriding.
+* `modifications` section stays almost the same as with the old policy except `jsonPatches` field.
+* `jsonPatches` is the alternative way of patching config which we also use in `ContainerPatch`.
+  It is more expensive to execute, because before applying the patch we need to marshal the config to JSON.
+  It should not be a default way of patching things, but this way we can overcome some limitations of a default proto merge. 
+  You can modify TypedConfig, because TypedConfig is serialized to JSON. This solves _Merging of TypedConfig_
+  You can modify parts of lists (add element, etc.). This solves _Appending to lists_
+
+#### Name alternatives
+
+We could argue that `Proxy Template` may not be the most accurate name. After all, we are not providing template, but we are altering Envoy Config.
+
+A couple of name alternatives:
+* MeshEnvoyConfig
+* MeshXDSConfig
+* MeshEnvoyMods (but the policy can potentially include other stuff than modifications in the future)
+* MeshProxyConfig (too generic? sounds like configuration of Kuma DP)
+* MeshProxyPatch
+
+While `MeshProxyPatch` seems to be the most accurate name, I'm hesitant to make this change. I think that changing the name may confuse existing users. 
+
+#### Other resources
+
+##### Transport Socket modifications
+
+We could introduce "native" transport_socket modification, but if we introduce `jsonPatches` we can also rely on it.
```

Comment:
So are we going to do this? Or `jsonPatch` is enough?

Reply:
We may do it, in the meantime relying on jsonPatch is enough